addons/l10n_in_ewaybill_stock/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,603 @@
+
+            if (
+                ewaybill.picking_type_code == 'dropship' and
+                (dest_partner := ewaybill.picking_id._get_l10n_in_dropship_dest_partner())
+            ):
+                ewaybill.partner_ship_to_id = dest_partner
+                ewaybill.partner_ship_from_id = ewaybill.picking_id.partner_id
+
+    @api.depends('fiscal_position_id')
+    def _compute_is_fiscal_position_invisible(self):
+        """
+        Make fiscal field invisible if a Sales Order (SO) or Purchase Order (PO) exists.
+
+        This function checks the presence of a Sales Order (SO) or Purchase Order (PO) in the system.
+        If either a SO or PO exists, the fiscal field will be made invisible.
+        `_l10n_in_get_fiscal_position` will return False in this case.
+
+        However, if no SO or PO exists, the fiscal field will be visible.
+        If a fiscal position is not set on the SO or PO, the field will still be invisible,
+        returning an empty recordset.
+
+        Example:
+        - self.env['account.fiscal.position'] == False (output will be False)
+        - self.env['account.fiscal.position'] == account.fiscal.position() (output will be True)
+        """
+        for ewaybill in self:
+            fiscal_position = ewaybill.picking_id._l10n_in_get_fiscal_position()
+            ewaybill.is_fiscal_position_invisible = (
+                fiscal_position == self.env['account.fiscal.position'] or
+                bool(fiscal_position)
+            )
+
+    @api.depends('partner_bill_from_id', 'partner_bill_to_id')
+    def _compute_fiscal_position(self):
+        for ewaybill in self.filtered(lambda ewb: ewb.state == 'pending'):
+            ewaybill.fiscal_position_id = (
+                ewaybill.picking_id._l10n_in_get_fiscal_position() or
+                self.env['account.fiscal.position']._get_fiscal_position(
+                    ewaybill.picking_type_code == 'incoming' and ewaybill.partner_bill_from_id or
+                    ewaybill.partner_bill_to_id
+                )
+            )
+
+    @api.depends('partner_ship_from_id', 'partner_ship_to_id', 'partner_bill_from_id', 'partner_bill_to_id')
+    def _compute_is_editable(self):
+        for ewaybill in self:
+            is_incoming = ewaybill.picking_type_code == "incoming"
+            ewaybill.is_bill_to_editable = not is_incoming
+            ewaybill.is_bill_from_editable = is_incoming
+            ewaybill.is_ship_from_editable = is_incoming and ewaybill._is_overseas()
+            ewaybill.is_ship_to_editable = not is_incoming and not ewaybill._is_overseas()
+
+    def _compute_content(self):
+        for ewaybill in self:
+            ewaybill.content = base64.b64encode(json.dumps(ewaybill._ewaybill_generate_direct_json()).encode())
+
+    @api.depends('name', 'state')
+    def _compute_display_name(self):
+        for ewaybill in self:
+            ewaybill.display_name = ewaybill.state == 'pending' and _('Pending') or ewaybill.name
```

Comment:
To show Pending on the top (display_name) before the ewaybill is generated because we want ewaybill_id(from government response to be displayed as name)